hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/InsertIntoHoodieTableCommand.scala

```diff
@@ -243,6 +256,8 @@ object InsertIntoHoodieTableCommand {
         RECORDKEY_FIELD_OPT_KEY.key -> primaryColumns.mkString(","),
         PARTITIONPATH_FIELD_OPT_KEY.key -> partitionFields,
         PAYLOAD_CLASS_OPT_KEY.key -> payloadClassName,
+        ENABLE_ROW_WRITER_OPT_KEY.key -> enableBulkInsert.toString,
+        HoodieWriteConfig.COMBINE_BEFORE_INSERT_PROP.key -> isPrimaryKeyTable.toString, // if the table has primaryKey, enable the combine
```

Comment:
why the table has primaryKey, should enable the combine?

Reply:
Because we must guarantee the  uniqueness for p-k table. So we should combine the input for pk-table.

Reply:
even for pk-table, in insert mode, there is no need to combine the input, right?

Reply:
We need to ensure that data is unique for pk-table just like database, so I do combine for the input.

Reply:
yeah, I am also not sure on this. We will do a preCombine step for any INSERT in general. wondering why not let users take up the responsibility.

Reply:
Just like the upsert operation,  Hudi do the combine automatic， we can do this for the user too which is much friendly for our users.